lld/test/MachO/link-csu-object.s

```diff
@@ -2,7 +2,7 @@
 # RUN: mkdir -p %t
 # RUN: llvm-mc -filetype=obj -triple=x86_64-apple-darwin %p/Inputs/libhello.s -o %t/hello.o
 # RUN: llvm-mc -filetype=obj -triple=x86_64-apple-darwin %s -o %t/main.o
-# RUN: %lld -L %t %t/main.o %t/hello.o -o %t/a.out
+# RUN: %lld -L %t %t/main.o -lhello.o -o %t/a.out
```

Comment:
I think @smeenai was suggesting  sth like:

`%lld -lhello %t/main.o -o %t/a.out` (ie., using `-l<lib_name>` syntax - __without__ the `.o`). 

Perhaps for easy comparison, I'd suggest leaving your line #5  alone - and add an additional link that use the `-l<lib_name>` syntax - then smoke-check that the two output are the same.
(@smeenai thoughts?)

Reply:
The point of the pr was for Csu support files, which specify the whole name of the object, `-lhello` wont link hello.o

This fixes building with MacOS X 10.7 and 10.6 SDKs which require linking with files like crt1.o, clang passes `-lcrt1.o` to the linker, which works with ld64 but not lld

Reply:
D'oh . Thanks !